indico/modules/rb/client/js/common/rooms/edit/RoomEditOptions.jsx

```diff
@@ -0,0 +1,162 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import {Dropdown, Form, Header, Tab} from 'semantic-ui-react';
+import React, {useCallback, useMemo} from 'react';
+import _ from 'lodash';
+import {FieldArray} from 'react-final-form-arrays';
+import PropTypes from 'prop-types';
+import shortid from 'shortid';
+import {Translate} from 'indico/react/i18n';
+import {FinalCheckbox, FinalField, FinalInput} from 'indico/react/forms';
+import EquipmentList from './EquipmentList';
+import DailyAvailability from './DailyAvailability';
+import NonBookablePeriods from './NonBookablePeriods';
+
+function RoomEditOptions({active, showEquipment, globalAttributes}) {
+  const attributeTitles = useMemo(
+    () => (globalAttributes ? _.fromPairs(globalAttributes.map(x => [x.name, x.title])) : []),
+    [globalAttributes]
+  );
+  const withKeyAttribute = useCallback(
+    value => value.map(e => ({...e, key: shortid.generate()})),
+    []
+  );
+
+  return (
+    <Tab.Pane active={active}>
+      <Header>
+        <Translate>Availability</Translate>
+      </Header>
+      <Form.Group>
+        <FinalField
+          name="bookable_hours"
+          component={DailyAvailability}
+          isEqual={_.isEqual}
+          format={value => (value ? [] : withKeyAttribute(value))}
+          hideErrorPopup={!active}
+        />
+        <FinalField
+          name="nonbookable_periods"
+          component={NonBookablePeriods}
+          isEqual={_.isEqual}
+          format={value => (value ? [] : withKeyAttribute(value))}
+          hideErrorPopup={!active}
+        />
+      </Form.Group>
+      <Header>
+        <Translate>Equipment and custom attributes</Translate>
+      </Header>
+      <Form.Group>
+        {showEquipment && (
+          <FinalField
+            name="available_equipment"
+            component={EquipmentList}
+            isEqual={_.isEqual}
+            componentLabel={Translate.string('Add new equipment')}
+            hideErrorPopup={!active}
+          />
+        )}
+        <FieldArray name="attributes" isEqual={_.isEqual}>
+          {({fields}) => {
+            if (!fields.value) {
+              return null;
+            }
+            const fieldsByName = fields.map(x => x.name);
+            const options = globalAttributes
+              .map(x => ({
+                key: x.name,
+                text: x.title,
+                value: x.name,
+              }))
+              .filter(attr => !fieldsByName.includes(attr.key));
```

Comment:
This doesn't work correctly; I can still add an attribute which is already defined.